hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/OzoneKMSUtil.java

```diff
@@ -128,6 +128,9 @@ public static URI getKeyProviderUri(UserGroupInformation ugi,
 
   public static KeyProvider getKeyProvider(final Configuration conf,
       final URI serverProviderUri) throws IOException{
+    if (serverProviderUri == null) {
```

Comment:
We should not call getKeyProvider when provider uri is not defined. In RpcClient#getKeyProvider() and RestClient#getkeyProvider(), we should do

String kpUri = getKeyProviderUri();
return kpUri == null ? null : OzoneKMSUtil.getKeyProvider(conf, kpUri);